osu.Game/Screens/Edit/Compose/Components/ComposeSelectionBox.cs

```diff
@@ -0,0 +1,352 @@
+            }
+        }
+
+        private bool canScaleX;
+
+        public bool CanScaleX
+        {
+            get => canScaleX;
+            set
+            {
+                canScaleX = value;
+                recreate();
+            }
+        }
+
+        private bool canScaleY;
+
+        public bool CanScaleY
+        {
+            get => canScaleY;
+            set
+            {
+                canScaleY = value;
+                recreate();
+            }
+        }
+
+        public const float BORDER_RADIUS = 3;
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RelativeSizeAxes = Axes.Both;
+
+            recreate();
+        }
+
+        private void recreate()
+        {
+            if (LoadState < LoadState.Loading)
+                return;
+
+            InternalChildren = new Drawable[]
+            {
+                new Container
+                {
+                    Masking = true,
+                    BorderThickness = BORDER_RADIUS,
+                    BorderColour = colours.YellowDark,
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+
+                            AlwaysPresent = true,
+                            Alpha = 0
+                        },
+                    }
+                },
+            };
+
+            if (CanRotate)
+            {
+                const float separation = 40;
+
+                AddRangeInternal(new Drawable[]
+                {
+                    new Box
+                    {
+                        Colour = colours.YellowLight,
+                        Blending = BlendingParameters.Additive,
+                        Alpha = 0.3f,
+                        Size = new Vector2(BORDER_RADIUS, separation),
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.BottomCentre,
+                    },
+                    new RotationDragHandle
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Y = -separation,
+                        HandleDrag = e => OnRotation?.Invoke(e),
+                        OperationStarted = operationStarted,
+                        OperationEnded = operationEnded
+                    }
+                });
+            }
+
+            if (CanScaleY)
+            {
+                AddRangeInternal(new[]
+                {
+                    new DragHandle
+                    {
+                        Anchor = Anchor.TopCentre,
+                        HandleDrag = e => OnScaleY?.Invoke(e, Anchor.TopCentre),
+                        OperationStarted = operationStarted,
+                        OperationEnded = operationEnded
+                    },
```

Comment:
These could be DRY'd a little. The only differing parameter is the anchor - some bit-masking should make it so that this code has to only be written once but only called 8 times with different anchors.

Reply:
yep, definitely. kinda sad you got to review this PR before I didn't a quality pass on it, shouldn't have PR'd yet 😅 .